packages/ai/core/tool/mcp/mcp-client.ts

```diff
@@ -119,6 +124,11 @@ class MCPClient {
       await this.transport.start();
       this.isClosed = false;
 
+      if (this.transport.sessionId) {
+        // We do not need to re-initialize if sessionId is already set:
+        return this;
+      }
```

Comment:
Is there a unit test that covers this behavior? Is there anything we need to do regarding the transport (e.g. will it reopen automatically)?

Reply:
adding! the behavior here is that after old connection is closed, a new transport is initialized and started w/ session id and no initialization request is sent to the server